### PR TITLE
Added class Connection.Util and ManufacturerJdbcDaoImpl

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                + "manufacturer_country) VALUE (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement createdStatement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            createdStatement.setString(INDEX_ONE, manufacturer.getName());
+            createdStatement.setString(INDEX_TWO, manufacturer.getCountry());
+            createdStatement.execute();
+            ResultSet resultSet = createdStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getLong(INDEX_ONE));
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java]: be attentive with: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/05_JBDC_checklist.html

File: src/main/resources/init_db.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE `taxi_service`.`manufacturers` (
```
Comment (Max-Pochepets) [inline src/main/resources/init_db.sql:2]: is the `schema` being created after this set of instructions?

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    private static final int INDEX_TWO = 2;
+    private static final int INDEX_THREE = 3;
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (manufacturer_name, "
+                + "manufacturer_country) VALUE (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement createdStatement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java]:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
            PreparedStatement createdStatement = connection
                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```
and then we can remove `statement.close()`

File: src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerJdbcDaoImpl implements ManufacturerDao {
+    private static final int INDEX_ONE = 1;
+    private static final int INDEX_TWO = 2;
+    private static final int INDEX_THREE = 3;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java]: особливого значення дані константи не несуть. Якщо в нас буде наприклад 30 полів в модельці, доведеться робити 30 констант. Тому пропоную видалити їх

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java [hunk 5]
```diff
@@ -15,25 +15,21 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createdStatement = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createdStatement.setString(1, manufacturer.getName());
+            createdStatement.setString(2, manufacturer.getCountry());
             createdStatement.execute();
             ResultSet resultSet = createdStatement.getGeneratedKeys();
             if (resultSet.next()) {
-                manufacturer.setId((Long)resultSet.getObject(INDEX_ONE));
+                manufacturer.setId(resultSet.getLong(1));
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java]: be attentive with checklist - https://mate-academy.github.io/jv-program-fulltime/02_jdbc/homework/05_JDBC_HW.html

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java (hunk 5 above)
Comment (vicire) [reply src/main/java/core/basesyntax/dao/ManufacturerJdbcDaoImpl.java]: yes, I saw this. Sorry, totally forgot that we can return Object of Long.class (Simple cast to Long, as I write earlier, didn`t work)

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**